src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -178,6 +178,24 @@ internal JsonConverter EffectiveConverter
         internal MemberTypes MemberType { get; private protected set; }
         internal bool IsVirtual { get; private set; }
 
+        /// <summary>
+        /// Provides information if property is initialized when constructor is called
+        /// </summary>
+        /// <remarks>
+        /// This property is modified after property's Configure but before its parent's JsonTypeInfo Configure is finished.
+        /// This is due to parameterized constructor logic needing propeties to be Configured.
+        /// </remarks>
+        internal bool IsConstructorParameter { get; private set; }
+
+        /// <summary>
+        /// Property is writable when it has a setter or is used as constructor parameter
+        /// </summary>
+        /// <remarks>
+        /// This property is modified after property's Configure but before its parent's JsonTypeInfo Configure is finished.
+        /// This is due to parameterized constructor logic needing propeties to be Configured.
+        /// </remarks>
+        internal bool IsWritable => CanDeserialize || IsConstructorParameter;
```

Comment:
`IsWritable` sounds kind of ambiguous when compared with `CanDeserialize`. Given that it's only being called in one place I think you could simply inline `CanDeserialize || IsConstructorParameter` to make it more clear what is being checked?